airflow/utils/sqlalchemy.py

```diff
@@ -153,6 +156,75 @@ def process_result_value(self, value, dialect):
         return BaseSerialization.deserialize(value)
 
 
+def sanitize_for_serialization(obj: V1Pod):
+    """
+    Convert pod to dict.... but *safely*.
+
+    When pod objects created with one k8s version are unpickled in a python
+    env with a more recent k8s version (in which the object attrs may have
+    changed) the unpickled obj may throw an error because the attr
+    expected on new obj may not be there on the unpickled obj.
+
+    This function still converts the pod to a dict; the only difference is
+    it populates missing attrs with None.
+
+    If obj is None, return None.
+    If obj is str, int, long, float, bool, return directly.
+    If obj is datetime.datetime, datetime.date
+        convert to string in iso8601 format.
+    If obj is list, sanitize each element in the list.
+    If obj is dict, return the dict.
+    If obj is OpenAPI model, return the properties dict.
+
+    :param obj: The data to serialize.
+    :return: The serialized form of data.
+
+    :meta private:
+    """
+    if obj is None:
+        return None
+    elif isinstance(obj, (float, bool, bytes, str, int)):
+        return obj
+    elif isinstance(obj, list):
+        return [sanitize_for_serialization(sub_obj) for sub_obj in obj]
+    elif isinstance(obj, tuple):
+        return tuple(sanitize_for_serialization(sub_obj) for sub_obj in obj)
+    elif isinstance(obj, (datetime.datetime, datetime.date)):
+        return obj.isoformat()
+
+    if isinstance(obj, dict):
+        obj_dict = obj
+    else:
+        obj_dict = {
+            obj.attribute_map[attr]: getattr(obj, attr)
+            for attr, _ in obj.openapi_types.items()
+            if getattr(obj, attr, None) is not None
+        }
```

Comment:
```suggestion
        attrs = (attr for attr in obj.openapi_types if hasattr(obj, attr))
        obj_dict = {
            obj.attribute_map[attr]: getattr(obj, attr)
            for attr in attrs
        }
```

Doe this work? (most relevant change is the `hasattr` use) I think the current `getattr` may incorrectly filter out None values.

Reply:
Can we not use https://github.com/apache/airflow/blob/main/airflow/serialization/serializers/kubernetes.py to not have duplication?

Reply:
That uses pod generator which uses the kubernetes client method that we are duplicating and fixing here

Reply:
I considered maybe just replacing that call with this logic but, I figure, probably best in general to just lean on kubernetes code and only vendor in and fix this but for backcompat when unpicking 

In face probably should also test that to dic fails before running through to reduce usage of it further

Reply:
> Doe this work? (most relevant change is the hasattr use) I think the current getattr may incorrectly filter out None values

i will try it @uranusjr

Reply:
actually, no @uranusjr filtering out None objects is intentional and the current behavior.  my only change is to tolerate when the attr is missing. that's all i want to change, otherwise it should be identical.